external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -3187,13 +3213,33 @@ fn match_pattern(context: &mut Context, sp!(loc, pat_): P::MatchPattern) -> E::M
             let tys = optional_types(context, pts_opt);
             match head_ctor_name {
                 Some(head_ctor_name @ sp!(_, EM::Variant(_, _))) => {
```

Comment:
This could could sweep the fields in a single pass:

```
let mut fields = vec![];
let mut ellipses = vec![];
for field in fields {
  match field {
      ....
  }
}
```